gobblin-yarn/src/main/java/gobblin/yarn/JobConfigurationManager.java

```diff
@@ -62,6 +63,7 @@ public JobConfigurationManager(EventBus eventBus, Optional<String> jobConfPackag
   protected void startUp() throws Exception {
     if (this.jobConfPackagePath.isPresent()) {
       File path = new File(this.jobConfPackagePath.get());
+      path = new File(path.getParent(), path.getName() + ".tar.gz");
```

Comment:
The config `gobblin.yarn.job.conf.dir` will specify a directory that contains the `.job` files. Then `StreamUtils.tar` will create a tarball with the name specified by the config + `.tar.gz` at the end. So the `ARCHIVE` added in the `GobblinYarnAppLauncher` is `config.getString(GobblinYarnConfigurationKeys.JOB_CONF_DIR_KEY) + .tar.gz`.

So in the AppMaster we need a append `.tar.gz` to the end of the name specified by `JOB_CONF_DIR_KEY`.